src/model_class.py

```diff
@@ -39,52 +38,45 @@ def list_image_files(self):
-
-                if (np.sum(np.isnan(image_array)) == 0) and (np.sum(np.isinf(image_array)) == 0):
-                    if model in ['ANN', 'RF']:  # for these two models append l=as tables
-                        datasets_list.append(image_array.reshape(-1, image_array.shape[-1]))
-                        targets_list.append(target_array.reshape(-1, target_array.shape[-1]))
-                    else:
-                        datasets_list.append(image_array)
-                        targets_list.append(target_array)
+        for image_filename, target_filename in list(zip(X, y)):
+            image_array = np.load(os.path.join(self.data_path, f'images/{subset}/', image_filename))
+            target_array = np.load(os.path.join(self.data_path, f'masks/{subset}/', target_filename))
+            if (np.sum(np.isnan(image_array)) == 0) and (np.sum(np.isinf(image_array)) == 0):
+                if model in ['ANN', 'RF']:  # for these two models append l=as tables
+                    datasets_list.append(image_array.reshape(-1, image_array.shape[-1]))
+                    targets_list.append(target_array.reshape(-1, target_array.shape[-1]))
                 else:
-                    print("isna", image_filename)
-            except:
-                print("doesn't exist", image_filename)
+                    datasets_list.append(image_array)
+                    targets_list.append(target_array)
+            else:
+                print("isna", image_filename)
         return datasets_list, targets_list
 
-    def train_validation_split(self, dataset, target_bin, fraction_val_set=0.2, model='ANN'):
-        logging.info('Splitting data into train and validation')
-        if model in ['ANN', 'RF']:  # for these two models append l=as tables
-            df_train, df_validation, target_bin_train, target_bin_validation = train_test_split(
-                dataset, target_bin, test_size=fraction_val_set, random_state=self.seed, stratify=target_bin
-            )
-        else:
-            all_indexes = np.arange(dataset.shape[0])
-            val_indexes = np.random.randint(0, dataset.shape[0], int(dataset.shape[0] * fraction_val_set))
-            train_indexes = np.array(list(set(all_indexes) - set(val_indexes)))
-            df_train = dataset[train_indexes, :, :, :]
-            target_bin_train = target_bin[train_indexes, :, :, :]
-            df_validation = dataset[val_indexes, :, :, :]
-            target_bin_validation = target_bin[val_indexes, :, :, :]
-        return df_train, df_validation, target_bin_train, target_bin_validation
+    def load_train_val_data(self, model):
+        """
+        load target and validation data from Dataset folder
+        :param model:
+        :return:
+        """
+        logging.info('reading processed images and masks.')
+        X_train, y_train, X_val, y_val = self.list_image_files()
+        size_train = 12
```

Comment:
Add a comment of this line that this is for check purposes.

Reply:
done